packages/field-base/src/input-control-mixin.js

```diff
@@ -23,6 +23,22 @@ export const InputControlMixin = (superclass) =>
   ) {
     static get properties() {
       return {
+        /**
+         * A pattern matched against individual characters the user inputs.
+         *
+         * When set, the field will prevent:
+         * - `keydown` events if the entered key doesn't match `/^allowedCharPattern$/`
+         * - `paste` events if the pasted text doesn't match `/^allowedCharPattern*$/`
+         * - `drop` events if the dropped text doesn't match `/^allowedCharPattern*$/`
+         *
+         * For example, to allow entering only numbers and minus signs, use:
+         * `allowedCharPattern = "[\\d-]"`
+         */
+        allowedCharPattern: {
```

Comment:
Should `allowedCharPattern` have the same visual feedback effect as `preventInvalidInput` does?

https://user-images.githubusercontent.com/1222264/172850854-66a24c8e-ac76-493e-8039-daa8bedf4df9.mp4

Reply:
Yes, thanks. I will move the corresponding logic to the mixin.

Reply:
If this is supposed to replace `preventInvalidInput`, then it probably should.

Reply:
Yes, good catch

Reply:
Done. I copied logic from `PatternMixin` instead of making it depend on `InputControlMixin` for simplicity.